gtk4/src/subclass/im_context.rs

```diff
@@ -506,6 +576,44 @@ unsafe extern "C" fn im_context_set_use_preedit<T: IMContextImpl>(
     imp.set_use_preedit(from_glib(use_preedit))
 }
 
+#[cfg(any(feature = "v4_2", docsrs))]
+unsafe extern "C" fn im_context_set_surrounding_with_selection<T: IMContextImpl>(
+    ptr: *mut ffi::GtkIMContext,
+    text: *const libc::c_char,
+    len: i32,
+    cursor_index: i32,
+    anchor_index: i32,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.imp();
+
+    imp.set_surrounding_with_selection(
+        &glib::GString::from_glib_none_num(text, len as usize),
```

Comment:
Why create a copy here, and not use e.g. `glib::GStr`?

Reply:
Also in other places here

Reply:
IIRC, we don't use glib::GStr in subclassing code at all. That is why I kept it using &str. But anyways, will update once #1678 lands

Reply:
It might be useful to use if otherwise you just have to convert to a `NUL`-terminated string again in the end. I don't know if that's the case here